### DISCUSSION TITLE
Model Viewer Editor is not working

### DISCUSSION THREAD

**Discussion by eddiecarbin** [google/model-viewer#2791] [General]
Model Viewer editor is no longer working with GLB models that I recently created and edited in the editor.  Has something recently changed with the editor?  Is there a better GLB editor out there?

https://modelviewer.dev/editor/

The same models load fine in the ThreeJS editor.  If there was an update with the editor is it possible for me to use a previous version of the editor?

**Comment by eddiecarbin** [comment]
This is the error I experience when I load any model into the editor. 

space-opera.js:93142 Uncaught (in promise) TypeError: Cannot destructure property 'type' of 'gltfMeshReference' as it is undefined.
    at new PrimitiveNode (space-opera.js:93142)

and this

space-opera.js:93152 Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'primitives')
    at new PrimitiveNode (space-opera.js:93152)

**Reply by elalish** [reply]
Yes, we've been updating the editor quite a bit lately and adding features; this sounds like a regression;  can you please link to  a non-working model so we can repro and get it fixed?

**Reply by elalish** [reply]
@timmmeh Any idea what  might be happening here?

**Reply by eddiecarbin** [reply]
All imported models are non-working besides the default built-in models.   I have tried to import random models from the web,  3dsmax Babylonjs exporter, ThreeJS  Glb files and models that were previously saved out from the editor itself.   

Is there a way to roll back to a previous version of the model-viewer editor while you guys fix the editor?

**Reply by elalish** [reply]
The models you've linked and everything else I've tried are working just fine for me. I'm on Chrome on OSX. What browser/OS are you using? Are there any console errors?
![image](https://user-images.githubusercontent.com/1649964/133657823-f3f1375d-76ff-4f1b-9521-cd0adf5c1712.png)

**Reply by eddiecarbin** [reply]
Yeah, I see that too but no animation, materials or model information. Did you click on Material or the Info Button?  And does it animate?    I am using windows and tested on chrome and firefox.

**Reply by elalish** [reply]
ah, thank you for the details! "not working" is not super specific...

Yes, I can repro; will have a fix out soon.

**Reply by eddiecarbin** [reply]
Yes, you are correct.   Sorry for the short description but you will find a list of items that aren't working with just a few navigation clicks.  Is your unit testing just seeing and turn the 3d model around?  Because none of the features in your editor work with the model.  It does look like a failed load so it breaks the interface.  Thanks though for looking at it.  Overall the editor has been an awesome tool and a time saver in development.   

Another good thing I noticed is that the GLB files that are saved out of the editor are considerably smaller than any other GLB file exporters.    Your editor is a must-have! :)

**Reply by elalish** [reply]
Thank you, glad it's helpful! It turns out this was actually a MV regression, so even more important. Part of why we serve the editor off master is to help get more testing on MV to ensure its releases  are very stable. And yes, as always, we need more automated testing. Turns out we didn't have a skinned model in our test suite, which is why  this slipped by.

**Comment by eddiecarbin** [comment]
The editor is working better.   It is still not able to export a model.  The download scene button throws an error.

Uncaught (in promise) Error: Failed to fetch url neutral
    at prepareZipArchive (space-opera.js:106686)
    at async HTMLElement.onDownloadClick (space-opera.js:106634)

**Reply by elalish** [reply]
Ugh, sorry, my bad. I'll get that fixed.

**Reply by eddiecarbin** [reply]
Thank you! It works great! I am able to finish this job.

**Comment by eddiecarbin** [comment]
Hi Sorry I have to post again.  But the mobile view with the model is not working either.   I am not able to send models to my phone like I was able to do earlier in the project.

**Reply by elalish** [reply]
Don't be sorry! This is super helpful, and I'm the one who's sorry. Can't believe I  overlooked this test too...

**Reply by eddiecarbin** [reply]
Thank you for understand and being so responsive.

**Reply by elalish** [reply]
Okay, that should be fixed now. However, that  code still  needs  some  work, so  I won't be  surprised if it  has some more issues. Please let me know  if you  find any.

**Reply by eddiecarbin** [reply]
yeah, I have a few issues. on the export of the model and have no clue what they might be just yet to report them.  I really wish for the last working build of the editor and that you do releases with the editor.  I am not doing this as a hobby and was almost done with a project and then you guys change the rules of the game.   I guess bad timing and I appreciate what you guys have created but I really don't want to be the tester for model-viewer-editor.

**Reply by elalish** [reply]
Yeah, that's fair. Thanks for  the feedback; it may be time to change our editor deployment process.

**Reply by eddiecarbin** [reply]
The issues I found with the export were with the original models.   All is well! Thank you for the help and you can view the project here.  

https://sites.google.com/view/dinosaursafari-jmdc/home
